include/pnet_api.h

```diff
@@ -1829,79 +1830,62 @@ typedef enum pnet_diag_ch_prop_dir_values
  * @param subslot             In:    The sub-slot.
  * @param ch                  In:    The channel number
- * @param ch_properties       In:    The channel properties (direction,
- *                                   data length, maintenance)
+ * @param is_channelgroup     In:    true if diag is related to a channel group.
+ *                                   Also known as "Accumulative".
+ *                                   Not yet used.
+ * @param ch_direction        In:    The channel direction
+ * @param ch_bits             In:    Number of bits in the channel
+ * @param severity            In:    Diagnosis severity
  * @param ch_error_type       In:    The channel error type.
  * @param ext_ch_error_type   In:    The extended channel error type
- *                                   (more details on the channel error type)
+ *                                   (more details on the channel error type).
+ *                                   Use 0 if not given.
  * @param ext_ch_add_value    In:    The extended channel error additional
- *                                   value (even more details).
- *                                   Only needed for standard format.
- * @param qual_ch_qualifier   In:    The qualified channel qualifier, describes
- *                                   severity. Only needed for standard format,
- *                                   only used if channel properties is
- *                                   MAINT_QUALIFIED.
+ *                                   value. Typically a measurement value (for
+ *                                   example number of dropped packets) related
+ *                                   to the error.
+ * @param qual_ch_qualifier   In:    More detailed severity information. Used
+ *                                   when severity =
+ *                                   PNET_DIAG_CH_PROP_MAINT_QUALIFIED
  *                                   Max one bit should be set.
- * @param usi                 In:    The USI. See above.
- * @param p_manuf_data        In:    The manufacturer specific diagnosis data.
- *                                   Only needed for manufacturer-specific
- *                                   diag format.
+ *                                   Set to 0 when not used.
  * @return  0  if the operation succeeded.
  *          -1 if an error occurred.
  */
-PNET_EXPORT int pnet_diag_add (
+PNET_EXPORT int pnet_diag_std_add (
    pnet_t * net,
    uint32_t arep,
    uint32_t api,
    uint16_t slot,
    uint16_t subslot,
    uint16_t ch,
-   uint16_t ch_properties,
+   bool is_channelgroup,
+   pnet_diag_ch_prop_dir_values_t ch_direction,
+   pnet_diag_ch_prop_type_values_t ch_bits,
+   pnet_diag_ch_prop_maint_values_t severity,
    uint16_t ch_error_type,
    uint16_t ext_ch_error_type,
    uint32_t ext_ch_add_value,
-   uint32_t qual_ch_qualifier,
-   uint16_t usi,
-   uint8_t * p_manuf_data);
+   uint32_t qual_ch_qualifier);
```

Comment:
This a a lot of arguments. Would if make it easier to define a diag struct and pass that to the operation? The application could define constant variables for the alarms it implements. Maybe special consideration for the add_value is needed since seems to be less static than the other variables.

Reply:
Implemented p`net_diag_source_t` containing six of the arguments.
Done